frontend/src/views/OrderView.vue

```diff
@@ -1,74 +1,183 @@
-<script setup>
+<script setup lang="ts">
+import AutoComplete from "primevue/autocomplete";
 import Button from "primevue/button";
+import Card from "primevue/card";
 import Dropdown from "primevue/dropdown";
 import FloatLabel from "primevue/floatlabel";
+import InputNumber from "primevue/inputnumber";
 import InputText from "primevue/inputtext";
 import Textarea from "primevue/textarea";
-import { ref } from "vue";
+import { computed, onMounted, ref } from "vue";
+import { useRoute, useRouter } from "vue-router";
 
+import { createOrder, getClients } from "@/backendClient";
 import { OrderStatus } from "@/global/types/appTypes";
+import type { OrderCreate, OrderUpdate } from "@/global/types/dataEditTypes";
+import type { Client } from "@/global/types/entities";
+import { ROUTES } from "@/router";
+
+interface ExtendedClient extends Client {
+  full_name?: string;
+}
+let orderInfo = ref<OrderUpdate | OrderCreate>({
+  client_id: "",
+  status: OrderStatus.preparation,
+  title: "",
+  description: "",
+  discount_percentage: 0,
+  discount_duration: 7,
+  can_have_cash_discount: false,
+});
 
 const orderStatusTypes = Object.values(OrderStatus);
-let status = ref(OrderStatus.preparation);
 
-const discountChoice = ["ja", "nein"];
-let discount = ref(discountChoice[0]);
+const discountChoice = [
+  { value: true, label: "ja" },
+  { value: false, label: "nein" },
+];
+const isEditing = computed(() => {
+  return Boolean(route.params.id);
+});
+const discountPeriodChoice = [7, 14];
+
+const selectedClient = ref<ExtendedClient>();
+
+const filteredClients = ref<ExtendedClient[]>([]);
+const clientsList = ref<ExtendedClient[]>([]);
+
+const isButtonDisabled = computed(() => {
+  if (orderInfo.value.title && selectedClient.value?.id && orderInfo.value.description) {
+    return false;
+  } else {
+    return true;
+  }
+});
+
+const searchClient = (event: any) => {
+  setTimeout(() => {
```

Comment:
This is (parts of) a debounce, right?

If we want to implement it at all, let's do it fully and create a helper method 👌🏻

Reply:
I will add it in the next pull request